packages/ui/src/working-groups/modals/CreateOpening/CreateOpening.tsx

```diff
@@ -0,0 +1,176 @@
+import BN from 'bn.js'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import styled from 'styled-components'
+
+import { useTransactionFee } from '@/accounts/hooks/useTransactionFee'
+import { useApi } from '@/api/hooks/useApi'
+import { ButtonPrimary, ButtonsGroup } from '@/common/components/buttons'
+import { DownloadButtonGhost } from '@/common/components/buttons/DownloadButtons'
+import { Modal, ModalHeader, ModalTransactionFooter } from '@/common/components/Modal'
+import {
+  /*StepDescriptionColumn, */ Stepper,
+  StepperModalBody,
+  StepperModalWrapper,
+} from '@/common/components/StepperModal'
+import { TextMedium } from '@/common/components/typography'
+import { useMachine } from '@/common/hooks/useMachine'
+import { useModal } from '@/common/hooks/useModal'
+import { SignTransactionModal } from '@/common/modals/SignTransactionModal/SignTransactionModal'
+import { isLastStepActive } from '@/common/modals/utils'
+import { getSteps } from '@/common/model/machines/getSteps'
+import { useYupValidationResolver } from '@/common/utils/validation'
+import { machineStateConverter } from '@/council/modals/AnnounceCandidacy/helpers'
+import { useMyMemberships } from '@/memberships/hooks/useMyMemberships'
+import { StyledStepperBody } from '@/proposals/modals/AddNewProposal'
+import { GroupIdToGroupParam } from '@/working-groups/constants'
+
+import { SuccessModal, CreateOpeningSteps as Steps, ImportOpening } from './components'
+import { createOpeningMachine, CreateOpeningMachineState, getTxParams } from './machine'
+import { OpeningConditions, CreateOpeningForm, CreateOpeningModalCall, OpeningSchema, defaultValues } from './types'
+
+export const CreateOpeningModal = () => {
+  const [showImport, setShowImport] = useState<boolean>(false)
+
+  const { api, connectionState } = useApi()
+  const { active: activeMember } = useMyMemberships()
+  const [state, send, service] = useMachine(createOpeningMachine)
+  const { hideModal, modalData } = useModal<CreateOpeningModalCall>()
+
+  const { group } = modalData
+  const workingGroupConsts = api?.consts[group]
+
+  const context = {
+    hiringTarget: 1,
+    minStake: new BN(50000),
+    group,
+    minUnstakingPeriodLimit: workingGroupConsts?.minUnstakingPeriodLimit,
+    minimumApplicationStake: workingGroupConsts?.minimumApplicationStake,
+  } as OpeningConditions
+  const path = useMemo(() => machineStateConverter(state.value), [state.value])
+  const resolver = useYupValidationResolver<CreateOpeningForm>(OpeningSchema, path)
+  const form = useForm<CreateOpeningForm>({ resolver, mode: 'onChange', defaultValues, context })
```

Comment:
Why is `defaultValues` needed here ?

Reply:
Because some parts of the form might have a default value which the user could change if they want to. like the Opening Duration field under Duration and Process